### PR TITLE
Refactor migration runner within transaction

### PR DESCRIPTION
**Pull request by Weiko**
Modifying the data-model can sometimes fail in the middle of your operation, due to the way we handle both metadata update and schema migration separately, a field can be created while the associated column creation failed (same for object/table and such). This is also an issue because WorkspaceMigrations are then stored as FAILED can never really recovered by themselves so the schema is broken and we can't update the models anymore.
This PR adds a executeMigrationFromPendingMigrationsWithinTransaction method where we can (and must) pass a queryRunner executing a transaction, which should come from the metadata services so that if anything during metadata update OR schema update fails, it rolls back everything (this also mean a workspaceMigration should never stay in a failed state now).
This also fixes some issues with migration not running in the correct order due to having the same timestamp and having to do some weird logic to fix that.

This is a first step and fix before working on a much more reliable solution in the upcoming weeks where we will refactor the way we interact with the data model.

### CODE REVIEW

**Comment by prastoin** [comment]
> (this also mean a workspaceMigration should never stay in a failed state now).

This means it will stay as no applied yet ?

File: packages/twenty-server/src/database/typeorm/typeorm.service.ts [hunk 1]
```diff
@@ -2,55 +2,23 @@ import { Injectable, OnModuleDestroy, OnModuleInit } from '@nestjs/common';
...
-import { User } from 'src/engine/core-modules/user/user.entity';
-import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
-import { AgentEntity } from 'src/engine/metadata-modules/agent/agent.entity';
 
 @Injectable()
 export class TypeORMService implements OnModuleInit, OnModuleDestroy {
   private mainDataSource: DataSource;
 
   constructor(private readonly twentyConfigService: TwentyConfigService) {
+    const isJest = process.argv.some((arg) => arg.includes('jest'));
```
Comment (prastoin) [inline packages/twenty-server/src/database/typeorm/typeorm.service.ts:12]: ***Concerns:*** Very insidious edge case where the filename of the file build contains jest within resulting in considering this is a test context
Would rather consume the `NODE_ENV` to `'test'` ?

File: packages/twenty-server/src/database/typeorm/typeorm.service.ts [hunk 2]
```diff
@@ -2,55 +2,23 @@ import { Injectable, OnModuleDestroy, OnModuleInit } from '@nestjs/common';
...
-        BillingCustomer,
-        BillingProduct,
-        BillingPrice,
-        BillingEntitlement,
-        PostgresCredentials,
-        WorkspaceSSOIdentityProvider,
-        ApprovedAccessDomain,
-        TwoFactorMethod,
-        AgentEntity,
+        `${isJest ? '' : 'dist/'}src/engine/core-modules/**/*.entity{.ts,.js}`,
```
Comment (prastoin) [inline packages/twenty-server/src/database/typeorm/typeorm.service.ts:20]: ***Praise:*** nice

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts [hunk 3]
```diff
@@ -266,44 +300,19 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
...
-          ],
-        );
-
-        await this.workspaceMigrationRunnerService.executeMigrationFromPendingMigrations(
-          updatedFieldMetadata.workspaceId,
-        );
-      }
-
-      await queryRunner.commitTransaction();
+      await this.workspaceMetadataVersionService.incrementMetadataVersion(
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:303]: ***Remark:*** Now incrementing the metadata before releasing the query, could this lead to cache hydration race condition ?

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts [hunk 4]
```diff
@@ -238,6 +242,36 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
...
+
+        await this.workspaceMigrationRunnerService.executeMigrationFromPendingMigrationsWithinTransaction(
+          updatedFieldMetadata.workspaceId,
+          queryRunner,
+        );
+      }
+
+      await queryRunner.commitTransaction();
+
       if (fieldMetadataInput.isActive === false) {
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:275]: ***Question:*** If the field get deactivated shoudn't we skip the above migrations ?

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts [hunk 5]
```diff
@@ -451,24 +457,35 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
...
+      });
+
+      await this.workspaceMetadataVersionService.incrementMetadataVersion(
+        workspaceId,
+      );
+
       return fieldMetadata;
     } catch (error) {
-      await queryRunner.rollbackTransaction();
+      if (queryRunner.isTransactionActive) {
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:482]: ***Hunch:*** Centralizing within an operation agnostic tooling that would handle error use case and releasing

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts [hunk 6]
```diff
@@ -943,26 +960,32 @@ export class FieldMetadataService extends TypeOrmQueryService<FieldMetadataEntit
...
 
-      await queryRunner.commitTransaction();
+      await this.workspaceMetadataVersionService.incrementMetadataVersion(
+        workspaceId,
+      );
 
       return createdFieldMetadatas;
     } catch (error) {
-      await queryRunner.rollbackTransaction();
+      if (queryRunner.isTransactionActive) {
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:982]: ***Question:*** From my undestanding, while creating many, if one of the create many creation fails, we will rollback the global transaction containing all the creation occurrences as we have only one transaction for all on them ?

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts [hunk 7]
```diff
@@ -30,15 +30,25 @@ export class IndexMetadataService {
...
+    workspaceId,
+    objectMetadata,
+    fieldMetadataToIndex,
+    isUnique,
+    isCustom,
+    indexType,
+    indexWhereClause,
+    queryRunner,
+  }: {
+    workspaceId: string;
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts:43]: ***Praise:*** I could fall in love for less

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts [hunk 8]
```diff
@@ -53,7 +63,11 @@ export class IndexMetadataService {
 
     let result: IndexMetadataEntity;
 
-    const existingIndex = await this.indexMetadataRepository.findOne({
+    const indexMetadataRepository = queryRunner
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts:66]: ***Question:*** Why for index metadata especially ? Shouldn't query runner be required here ?

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts [hunk 9]
```diff
@@ -186,23 +211,33 @@ export class IndexMetadataService {
...
-    isCustom: boolean,
-    indexType?: IndexType,
-    indexWhereClause?: string,
-  ) {
+  async createIndexCreationMigration({
+    workspaceId,
+    objectMetadata,
+    fieldMetadataToIndex,
+    isUnique,
+    _isCustom,
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts]: ***Nitpick:*** Could remove from destruct as unused

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts [hunk 10]
```diff
@@ -186,23 +211,33 @@ export class IndexMetadataService {
...
+    indexType,
+    indexWhereClause,
+    queryRunner,
+  }: {
+    workspaceId: string;
+    objectMetadata: ObjectMetadataEntity;
+    fieldMetadataToIndex: Partial<FieldMetadataEntity>[];
+    isUnique: boolean;
+    _isCustom: boolean;
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts]: ***Question:*** Should be documented as deprecated ?

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts [hunk 11]
```diff
@@ -30,15 +30,25 @@ export class IndexMetadataService {
...
-  async createIndexMetadata(
-    workspaceId: string,
-    objectMetadata: ObjectMetadataEntity,
-    fieldMetadataToIndex: Partial<FieldMetadataEntity>[],
-    isUnique: boolean,
-    isCustom: boolean,
-    indexType?: IndexType,
-    indexWhereClause?: string,
-  ) {
+  async createIndexMetadata({
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts:33]: ***Question:*** From my understanding as index migrations result from CRUD operations on field metadata or object metadata operations, we do not need to handle transaction rollback on failure as it handled by the caller ?

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts [hunk 12]
```diff
@@ -92,332 +95,416 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
...
-
-    validateObjectMetadataInputNamesOrThrow(objectMetadataInput);
-    validateObjectMetadataInputLabelsOrThrow(objectMetadataInput);
+    const mainDataSource =
+      await this.workspaceDataSourceService.connectToMainDataSource();
+    const queryRunner = mainDataSource.createQueryRunner();
+
+    await queryRunner.connect();
+    await queryRunner.startTransaction();
+    let createdObjectMetadata;
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts]: ***Nitpick:*** Unless I'm mistaken we could stick to the const and return the createdObjectMetadata only within the try block ? Does not make really sense to return a variable that in the end has been rolled back in case of an error ?

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts [hunk 13]
```diff
@@ -92,332 +95,416 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
...
-        existingObjectMetadataCombinedWithUpdateInput.namePlural,
-      );
-    }
+    const mainDataSource =
+      await this.workspaceDataSourceService.connectToMainDataSource();
+    const queryRunner = mainDataSource.createQueryRunner();
+
+    await queryRunner.connect();
+    await queryRunner.startTransaction();
+    let updatedObject;
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts]: ***Remark:*** Same than for createdObjectMetadata

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts [hunk 14]
```diff
@@ -92,332 +95,416 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
...
+        isActive: true,
+        isCustom: !objectMetadataInput.isRemote,
+        isSystem: false,
+        isRemote: objectMetadataInput.isRemote,
+        isSearchable: !objectMetadataInput.isRemote,
+        fields: objectMetadataInput.isRemote ? [] : baseCustomFields,
+        labelIdentifierFieldMetadataId,
+      });
+
+      if (objectMetadataInput.isRemote) {
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:193]: ***Concern:*** We should double check in production no user as one of them to true, might wanna also cover that new exception within an integration test

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts [hunk 15]
```diff
@@ -92,332 +95,416 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
...
+        });
+      }
+      validateMetadataIdentifierFieldMetadataIds({
+        fieldMetadataItems: Object.values(existingObjectMetadata.fieldsById),
+        labelIdentifierFieldMetadataId:
+          inputPayload.labelIdentifierFieldMetadataId,
+        imageIdentifierFieldMetadataId:
+          inputPayload.imageIdentifierFieldMetadataId,
+      });
+      updatedObject = await objectMetadataRepository.save({
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts]: ***Question:*** From update to save, and only for object metadata ? Unless I'm mistaken this could introduce higher risk of concurrency update race conditions ?

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts [hunk 16]
```diff
@@ -92,332 +95,416 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
...
+      );
 
-    const updatedObject = await super.updateOne(inputId, inputPayload);
+      await this.workspaceMigrationRunnerService.executeMigrationFromPendingMigrationsWithinTransaction(
+        workspaceId,
+        queryRunner,
+      );
+      if (inputPayload.labelIdentifierFieldMetadataId) {
+        const labelIdentifierFieldMetadata =
+          existingObjectMetadata.fieldsById[
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:362]: ***Concerns:*** existingObjectMetadata might contain a just added field to object metadata ?

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts [hunk 17]
```diff
@@ -489,49 +577,54 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
...
       if (
         (inputPayload.labelPlural || inputPayload.icon) &&
         (inputPayload.labelPlural !== existingObjectMetadata.labelPlural ||
           inputPayload.icon !== existingObjectMetadata.icon)
       ) {
-        await this.objectMetadataRelatedRecordsService.updateObjectViews(
-          objectMetadataForUpdate,
-          objectMetadataForUpdate.workspaceId,
-        );
+        return true;
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts]: ***Remark:*** It's quite tedious to understand what the caller would infer from the returned boolean
I would either:
- Extract this into a dedicated util
- make the function return a record that contains a meaningfully named flag
```ts
{
    didUpdateLabelOrIcon: boolean
}
```

File: packages/twenty-server/src/engine/metadata-modules/search-vector/search-vector.service.ts [hunk 18]
```diff
@@ -101,24 +106,31 @@ export class SearchVectorService {
...
-    await this.indexMetadataService.createIndexMetadata(
-      objectMetadataInput.workspaceId,
-      createdObjectMetadata,
-      [searchVectorFieldMetadata],
-      false,
-      false,
-      IndexType.GIN,
-    );
+    await this.indexMetadataService.createIndexMetadata({
+      workspaceId: objectMetadataInput.workspaceId,
```
Comment (prastoin) [inline packages/twenty-server/src/engine/metadata-modules/search-vector/search-vector.service.ts:113]: ***Praise:***  
<image src="https://github.com/user-attachments/assets/55bc8090-84c1-47c2-a344-7a23f4d3c11e" width=300 />

File: packages/twenty-server/src/engine/workspace-manager/workspace-migration-runner/workspace-migration-runner.service.ts [hunk 19]
```diff
@@ -32,6 +32,54 @@ export class WorkspaceMigrationRunnerService {
...
+      await transactionQueryRunner.query(
+        `UPDATE "core"."workspaceMigration" SET "appliedAt" = NOW() WHERE "id" = $1 AND "workspaceId" = $2`,
+        [parentMigrationId, workspaceId],
+      );
+    }
+
+    return migrationActionsWithParent.map((item) => item.tableAction);
+  }
+
   public async executeMigrationFromPendingMigrations(
```
Comment (prastoin) [inline packages/twenty-server/src/engine/workspace-manager/workspace-migration-runner/workspace-migration-runner.service.ts:83]: ***Question:*** Does this get deprecated in the best of the world ?

**Review by prastoin — COMMENTED** [review]
Wow that's cool ! Very good job, sorry for the comments/questions quantity
We can do some peer on them when you're at the office
Regarding testing, what would you think about adding a test, maybe on a `createMany` that insert an invalid creation and expect nothing the be created in an integration test ?
I don't know if it makes sense to implement such transaction coverage on each service.
I'm also afraid that this pattern is a bit too duplicated and could lead in the future to additions that do not follow the same pattern ( devX might not be enough "rigid" ) but in the end is it expected to change that much too..

Very nice overall !

File: packages/twenty-server/src/database/typeorm/typeorm.service.ts (hunk 1 above)
Comment (Weiko) [reply packages/twenty-server/src/database/typeorm/typeorm.service.ts:12]: That is very true, I blindly copy pasted the existing code from core.datasource.ts 🙂 It will probably won't happen (a file containing jest and injecting TypeORMService) but still I prefer to go with the NODE_ENV indeed!

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts (hunk 3 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:303]: Oops! No the increment should come after, good catch

Comment (prastoin) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:303]: Unless I'm mistaken it's the case for each updated services

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts (hunk 6 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:982]: indeed since our migration is declaring multiple fields at once for now

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts (hunk 10 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts]: I'll just remove it

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts (hunk 11 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts:33]: I'm not sure I understand this one. Wdym? field metadata changes result from CRUD operations on field metadata and sometimes you want to interact with index (by creating/destroying/recomputing them, depending on the action on the associated field) if the field migration fail you also want to rollback the change on the index, the caller didn't commit its change yet iirc

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts (hunk 8 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts:66]: I'm leaving it optional so you can still call the methods without a query runner and use the one that is injected in the class. Using a query runner here is almost only relevant when you want to work with transactions

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts (hunk 14 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:193]: isRemote has never really been launched, also here we are only throwing at creation step so it won't break existing logic if for some reason remote objects still exist.
This feature will probably come back later but I prefer to remove all the related code since it has not been used for a year and I'm not even sure it's still working and would prefer not to adapt the code again, especially since we will change those services pretty soon.

I should throw a 400 though!

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts (hunk 12 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts]: Ah, good point, there is nothing after the try block anymore now so I can do that

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts (hunk 13 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts]: Yes same answer 👍

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts (hunk 15 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts]: not sure about the race condition but you are right I should stick with the existing updateOne, not sure why I've changed that!

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts (hunk 16 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:362]: are you talking about a race condition here? Since we are using versioned cache map of object + a transaction I don't think this would happen (I believe this logic existed before as well)

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts (hunk 17 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts]: Sure, I can also find a better name for the method

File: packages/twenty-server/src/engine/workspace-manager/workspace-migration-runner/workspace-migration-runner.service.ts (hunk 19 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/workspace-manager/workspace-migration-runner/workspace-migration-runner.service.ts:83]: Probably yes, I didn't want to make too many changes in the existing code since we will have to go over all of this pretty soon, but in the end the thing that will run "schema migrations" will do that in a transaction as well yes

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts (hunk 4 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:275]: I don't think so, you can modify the field and deactivate it at the same time, at least the API allows it. Deactivating a field does not change the workspace schema

**Comment by Weiko** [comment]
> > (this also mean a workspaceMigration should never stay in a failed state now).
> 
> This means it will stay as no applied yet ?

it won't even be stored in the DB

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts (hunk 3 above)
Comment (Weiko) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:303]: in deleteOneField we correctly increment the version after the commit for example, I'll double check everything anyway

File: packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts (hunk 11 above)
Comment (prastoin) [reply packages/twenty-server/src/engine/metadata-modules/index-metadata/index-metadata.service.ts:33]: I meant that transaction rollback operation is the caller responsability

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts (hunk 4 above)
Comment (prastoin) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:275]: Ok !

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts (hunk 16 above)
Comment (prastoin) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:362]: Not a even a race condition, lets say you have a update on the object metadata that adds a fields, unless I'm mistaken it won't be referenced within the `existingObjectMetadata.fieldsById` ?

Comment (prastoin) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:362]: Unless I'm mistaken it wasn't
```ts
      const labelIdentifierFieldMetadata =
        await this.fieldMetadataRepository.findOneByOrFail({
          id: inputPayload.labelIdentifierFieldMetadataId,
          objectMetadataId: inputId,
          workspaceId: workspaceId,
        });
```

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts (hunk 3 above)
Comment (prastoin) [reply packages/twenty-server/src/engine/metadata-modules/field-metadata/field-metadata.service.ts:303]: My bad though it was before a commit not a release ! RAS

File: packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts (hunk 16 above)
Comment (prastoin) [reply packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts:362]: In fact field metadata creation is an outside operation, maybe on connect import this might become an issue but indeed you cannot add ftm a field to an object metadata through an udpateObjectMetadata anw

**Review by prastoin — APPROVED** [review]
<image src=https://github.com/user-attachments/assets/ce164c8d-a56f-46bb-afad-13430c7e0530  width=300 />

**Review by charlesBochet — APPROVED** [review]

**Comment by prastoin** [comment]
> ## Suspect Issues
> This pull request was deployed and Sentry observed the following issues:
> 
> * ‼️ [**TypeORMError: Column "contactType" was not found i...**](https://twenty-v7.sentry.io/issues/6726874188/?referrer=github-pr-bot) in `prod`
> 
> Did you find this useful? React with a 👍 or 👎

Impossible this hasn't been released in production yet, this will fix the issue
